### DISCUSSION TITLE
Does the ``env`` or the ``robots`` has pre-defined low level actions such as ``move_forward``, ``turn_right``, ``turn_left``?

### DISCUSSION THREAD

**Discussion by Yonggie** [StanfordVL/BEHAVIOR-1K#948] [Q&A] [answered]
I see nearly every example in navigation are all using the ``env.action_space.sample()`` to generate actions. But what if I just need the specific actions?

Hope I can find help!

**Comment by hang-yin** [comment]
Hi @Yonggie, we do have some experimental primitive actions implemented. If you go to the `og-develop` branch, you will see a `test_robot_load_drive` test case in `tests/test_robot_states_flatcache.py`. In there, you will see examples of how you can use the starter action primitives for simple navigation and manipulation. Let me know if you run into any problems!

**Comment by hang-yin** [comment, marked as answer]
Hi @Yonggie, with our most recent release, you can try out our action primitives. You will need `OMPL` to use them so just start with the Docker container. There are examples in the `examples/action_primitives` directory, and you can find out more about the full set of primitives in the `omnigibson/action_primitives` directory. We have a physically simulated set and a symbolic, teleporting set.

**Reply by Yonggie** [reply]
Will try, thanks!

**Reply by Yonggie** [reply]
Hi, thanks for the efforts. After several experiments, We found the ``action primitives``(toggle_on, grasp, etc...) scripts are demo-like and sometimes does not work well.
